source/fuzz/transformation_swap_commutable_operands.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "source/fuzz/transformation_swap_commutable_operands.h"
+
+#include "source/fuzz/fuzzer_util.h"
+#include "source/fuzz/instruction_descriptor.h"
+
+namespace spvtools {
+namespace fuzz {
+
+TransformationSwapCommutableOperands::TransformationSwapCommutableOperands(
+  const spvtools::fuzz::protobufs::TransformationSwapCommutableOperands& message
+) : message_(message) {}
+
+TransformationSwapCommutableOperands::TransformationSwapCommutableOperands(
+  const protobufs::InstructionDescriptor& instruction_descriptor
+) {
+  *message_.mutable_instruction_descriptor() = instruction_descriptor;
+}
+
+bool TransformationSwapCommutableOperands::IsApplicable(
+  opt::IRContext* /*unused*/,
+  const spvtools::fuzz::FactManager& /*unused*/
+) const {
+  SpvOp opcode = static_cast<SpvOp>(message_.instruction_descriptor().target_instruction_opcode());
+  return spvOpcodeIsCommutative(opcode);
```

Comment:
This needs a bit more work: you should check that the instruction descriptor refers to an instruction that actually exists (find the instruction and check that the pointer you get is not null, returning false if so).

As a sanity check you could also assert that the located instruction has the same opcode as in the descriptor.

Reply:
Done.